config/config-api/src/main/java/com/thoughtworks/go/config/Cacheable.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.config;
+
+import static com.thoughtworks.go.util.CachedDigestUtils.sha256Hex;
+
+public interface Cacheable {
+    default String etag() {
+        return sha256(Integer.toString(hashCode()));
+    }
+
+    default String sha256(String subject) {
```

Comment:
We end up allowing subclasses to override this method, which doesn't seem right to me. Maybe just inline it?

Reply:
@akshaydewan are you referring to making this not default (i.e., abstract)? We could do that, but I think a sha256 of the hashcode is a sensible default, so long as you implement `hashCode()`. Many of our models do.

If you’ve strong objections, I’m happy to inline that and leave the method abstract in the `Cacheable` interface.

Reply:
I'll rephrase - From an interface point of view, having a `Cacheable` object implement/override `etag()` seems reasonable, but having  it implement/override `sha256` seems like a smell to me. It looks as though we're using `sha256` more like a utility function in the interface, in which case, we can call `com.thoughtworks.go.util.CachedDigestUtils.sha256Hex` directly. Or am I missing something?

Reply:
Ah I get what you meant. Yeah I can certainly see your point. I mostly put it there for convenience for any other class that wants to implement cacheable. But I probably should have Inlined it instead since it’s already a utility method.